### DISCUSSION TITLE
Unable to restore user from cookie with cypress-data-session

### DISCUSSION THREAD

**Discussion by NirShihor** [cypress-io/cypress#22591] [Questions and Help]
Would anyone familiar with cypress-data-session library know why this is not working?
The set-up for the user works insofar as it goes through the signin process and into the account page, with tests passing. 
However, on consequent runs, once the user has been created and stored, the recreate section fails and the page opens again at the signin. 

`

beforeEach(function () {
  const userEmail = this.Email.toString()
  const userPassword = '12qwER£$'
  cy.dataSession({
    name: 'User',
    setup() {
      cy.visit('/account/signin')
      cy.get('#email').type(userEmail)
      cy.get('#password').type(userPassword)
      cy.get('[data-testid=signin]').should('be.visible').trigger('mouseover').click({ force: true })
      cy.location('pathname').should('eq', '/account/')
      cy.getCookie('builderSessionId')
    },
    validate: true,
    recreate(cookie) {
      cy.setCookie('builderSessionId', cookie.value)
      cy.visit('/account/')
    },
    shareAcrossSpecs: true
  })
})
`

**Comment by stavares843** [comment]
hello! 👋 

have you tried [this](https://www.npmjs.com/package/cypress-localstorage-commands) plugin to keep the whole localStorage between Cypress tests?

**Comment by NirShihor** [comment]
I need to retain session across spec files, not just tests.   
  
Thanks,
  
Nir
  
  
  
  
  
  
  

  
  
>
